app/uk/gov/hmrc/gform/sharedmodel/formtemplate/FormatExpr.scala

```diff
@@ -208,10 +282,114 @@ case object ChildBenefitNumber extends TextConstraint
 object TextConstraint {
   val defaultWholeDigits = 11
   val defaultFactionalDigits = 2
+  val defaultDisplayWidthSize = 20
+  val defaultDisplayWidthXsSizeForText = 4
+  val defaultDisplayWidthSSizeForText = 5
+  val defaultDisplayWidthMSizeForText = 10
+  val defaultDisplayWidthLSizeForText = 20
+  val defaultDisplayWidthXlSizeForText = 30
+  val defaultDisplayWidthXxlSizeForText = 40
+  val defaultDisplayWidthXsSizeForNumber = 2
+  val defaultDisplayWidthSSizeForNumber = 3
+  val defaultDisplayWidthMSizeForNumber = 4
+  val defaultDisplayWidthLSizeForNumber = 5
+  val defaultDisplayWidthXlSizeForNumber = 10
+  val defaultDisplayWidthXxlSizeForNumber = 20
+  val defaultSizeNumber = 10
+  val defaultSizePositiveNumber = 10
+  val defaultSizeBasicText = 20
+  val defaultSizeShortText = 20
+  val defaultSizeLookup = 20
+  val defaultSizeTextWithRestrictions = 20
+  val defaultSizeSterling = 10
+  val defaultSizeUkBankAccountNumber = 10
+  val defaultSizeUkSortCodeFormat = 2
+  val defaultSizeSubmissionRefFormat = 20
+  val defaultSizeTelephoneNumber = 20
+  val defaultSizeEmail = 30
+  val defaultSizeEmailVerifiedBy = 10
+  val defaultSizeUTR = 10
+  val defaultSizeNINO = 20
+  val defaultSizeUkVrn = 10
+  val defaultSizeCountryCode = 5
+  val defaultSizeNonUkCountryCode = 5
+  val defaultSizeCompanyRegistrationNumber = 20
+  val defaultSizeEORI = 20
+  val defaultSizeUkEORI = 20
+  val defaultSizeChildBenefitNumber = 20
 
   implicit val format: OFormat[TextConstraint] = derived.oformat[TextConstraint]
 
   def filterNumberValue(s: String): String = s.filterNot(c => (c == '£'))
+
+  def getSizeForDisplayWidthForText(displayWidth: DisplayWidth.DisplayWidth): Int = displayWidth match {
+    case DisplayWidth.XS  => TextConstraint.defaultDisplayWidthXsSizeForText
+    case DisplayWidth.S   => TextConstraint.defaultDisplayWidthSSizeForText
+    case DisplayWidth.M   => TextConstraint.defaultDisplayWidthMSizeForText
+    case DisplayWidth.L   => TextConstraint.defaultDisplayWidthLSizeForText
+    case DisplayWidth.XL  => TextConstraint.defaultDisplayWidthXlSizeForText
+    case DisplayWidth.XXL => TextConstraint.defaultDisplayWidthXxlSizeForText
+  }
+
+  def getSizeForDisplayWidthForNumber(displayWidth: DisplayWidth.DisplayWidth): Int = displayWidth match {
+    case DisplayWidth.XS  => TextConstraint.defaultDisplayWidthXsSizeForNumber
+    case DisplayWidth.S   => TextConstraint.defaultDisplayWidthSSizeForNumber
+    case DisplayWidth.M   => TextConstraint.defaultDisplayWidthMSizeForNumber
+    case DisplayWidth.L   => TextConstraint.defaultDisplayWidthLSizeForNumber
+    case DisplayWidth.XL  => TextConstraint.defaultDisplayWidthXlSizeForNumber
+    case DisplayWidth.XXL => TextConstraint.defaultDisplayWidthXxlSizeForNumber
+  }
```

Comment:
Just out of curiosity, why not to do simple:
```
  def getSizeForDisplayWidthForText(displayWidth: DisplayWidth.DisplayWidth): Int = displayWidth match {
    case DisplayWidth.XS  => 4
    case DisplayWidth.S   => 5
    case DisplayWidth.M   => 10
    case DisplayWidth.L   => 20
    case DisplayWidth.XL  => 30
    case DisplayWidth.XXL => 40
  }

  def getSizeForDisplayWidthForNumber(displayWidth: DisplayWidth.DisplayWidth): Int = displayWidth match {
    case DisplayWidth.XS  => 2
    case DisplayWidth.S   => 3
    case DisplayWidth.M   => 4
    case DisplayWidth.L   => 5
    case DisplayWidth.XL  => 10
    case DisplayWidth.XXL => 20
  }
```
?

It is more readable (imho) and much less typing.

Reply:
I wanted to define them as constants. At the moment, these constants are used at only one place. In future, they may be used in other places. And if any changes in them will be reflected in all used places and avoid defects.